demo/src/components/SrlComponent.js

```diff
@@ -34,6 +34,77 @@ const description = (
   </div>
 );
 
+function getStrIndex(words, wordIdx) {
+  if (wordIdx < 0) throw new Error(`Invalid word index: ${wordIdx}`);
+  return words.slice(0, wordIdx).join(' ').length;
+}
+
+function toHierplaneTrees(response) {
+  const text = response.words.join(' ');
+
+  // We create a tree for each verb
+  const trees = response.verbs.map(({ verb, tags }) => {
+    const verbTagIdx = tags.findIndex(tag => tag === 'B-V');
+    const start = getStrIndex(response.words, verbTagIdx);
+
+    const ignoredSpans = tags.reduce((children, tag, idx) => {
+      if (tag === 'O') {
+        const word = response.words[idx];
+        const start = getStrIndex(response.words, idx);
+        const end = start + word.length + 1;
+        const child = {
+          spanType: 'ignored',
+          start,
+          end
+        };
+        children.push(child);
+      }
+      return children;
+    }, []);
+
+    return {
+      text,
+      root: {
+        word: verb,
+        nodeType: 'V',
+        attributes: [ 'VERB' ],
+        spans: [{
+          start,
+          end: start + verb.length + 1,
+        }, ...ignoredSpans],
+        children: tags.reduce((children, tag, idx) => {
+          if (tag !== 'B-V' && tag.startsWith('B-')) {
+            const word = response.words[idx];
+            const link = tag.split('-').pop().replace(/\d+/g, '');
```

Comment:
The numeric characters in the arguments actually carry some meaning for SRL - can you leave them in?

Reply:
Yup, for sure!